devito/finite_differences/derivative.py

```diff
@@ -293,15 +300,32 @@ def _xreplace(self, subs):
             except AttributeError:
                 return new, True
 
+        # Resolve nested derivatives
+        dsubs = {k: v for k, v in subs.items() if isinstance(k, Derivative)}
+        expr = self.expr.xreplace(dsubs)
+
         subs = self._ppsubs + (subs,)  # Postponed substitutions
-        return self._rebuild(subs=subs), True
+        return self._rebuild(subs=subs, expr=expr), True
 
     @cached_property
     def _metadata(self):
         ret = [self.dims] + [getattr(self, i) for i in self.__rkwargs__]
         ret.append(self.expr.staggered or (None,))
         return tuple(ret)
 
+    def filter_dims(self, col, as_tuple=False, neg=False):
+        """
+        Filter collectiion to only keep the derivative's dimensions as keys.
```

Comment:
typo collection

also, "derivative" -> "Derivative"

again, fixable in abother branch